scripts/open-coredump.sh

```diff
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+# Script for "one-click" opening of coredumps.
+# It extracts the build-id from the coredump, retrieves metadata for that
+# build, downloads the binary package, the source code and finally
+# launches the dbuild container, with everything ready to load the coredump.
+# The script is reentrant: running it after the prepartory steps will re-use what
+# is already donwloaded. It is *strongly* recommended to run this from an empty
+# directory, with nothing but the core-file present.
+#
+# Requirements:
+# * eu-unstrip
+# * curl
+# * jq
+# * git
+
+set -e
+
+SCYLLA_S3_RELOC_SERVER_URL="http://backtrace.scylladb.com"
+
+if [[ $# != 1 ]]; then
+	echo Please provide the path to the coredump
+	exit 1
+fi
+
+for required in eu-unstrip jq curl git; do
+	if ! type $required >& /dev/null; then
+		echo Please install $required first
+		exit 1
+	fi
+done
+
+COREFILE=$1
+
+BUILD_ID=$(eu-unstrip -n --core ${COREFILE} | cut -f2 -d' ' | cut -f1 -d@)
+
+echo "Build id: ${BUILD_ID}"
+
+BUILD=$(curl -s -X GET "${SCYLLA_S3_RELOC_SERVER_URL}/build.json?build_id=${BUILD_ID}")
+
+RESPONSE_BUILD_ID=$(jq -r .build_id <<< $BUILD)
+VERSION=$(jq -r .version <<< $BUILD)
+PRODUCT=$(jq -r .product <<< $BUILD)
+RELEASE=$(jq -r .release <<< $BUILD)
+ARCH=$(jq -r .arch <<< $BUILD)
+BUILD_MODE=$(jq -r .build_mode <<< $BUILD)
+PACKAGE_URL=$(jq -r .package_url <<< $BUILD)
+
+if [[ "$RESPONSE_BUILD_ID" != "$BUILD_ID" ]]
+then
+    echo "Mismatching build id: requested ${BUILD_ID} but got ${RESPONSE_BUILD_ID}";
+    exit 1
+fi
+
+echo "Matching build is ${PRODUCT}-${VERSION} ${RELEASE} ${BUILD_MODE}-${ARCH}"
+
+PACKAGE_FILE=$(basename ${PACKAGE_URL})
+
+if ! [[ -d ${PRODUCT} ]]
+then
+    if ! [[ -f ${PACKAGE_FILE} ]]
+    then
+        echo "Downloading relocatable package from ${PACKAGE_URL}"
+        curl --output ./${PACKAGE_FILE} ${PACKAGE_URL}
+    else
+        echo "Relocatable package ${PACKAGE_URL} already downloaded"
+    fi
+
+    echo "Extracting package ${PACKAGE_FILE}"
+    tar -zxf $PACKAGE_FILE
+else
+    echo "Relocatable package ${PACKAGE_URL} already downloaded and extracted"
+fi
+
+COMMIT_HASH=$(cut -f3 -d. <<< $RELEASE)
+BASE_VERSION=$(grep -o "^[0-9]\+\.[0-9]\+" <<< $VERSION)
+BRANCH=branch-${BASE_VERSION}
+
+if ! [[ -d ${PRODUCT}.git ]]
+then
+    echo "Cloning ${PRODUCT}.git"
+    git clone -b ${BRANCH} https://github.com/scylladb/${PRODUCT}.git ${PRODUCT}.git
+    cd ${PRODUCT}.git
+    git checkout -q ${COMMIT_HASH}
+    # Skip the other submodules, they are not needed for debugging
+    git submodule update -q --depth=1 --init seastar
+    cd ..
+else
+    echo "${PRODUCT}.git already cloned"
+fi
+
+if ! [[ -f scylla-gdb.py ]]
+then
+    MAIN_BRANCH=master
+    if [[ "${PRODUCT}" == "scylla-enterprise" ]]
+    then
+        MAIN_BRANCH=enterprise
+    fi
+
+    echo "Downloading scylla-gdb.py"
+    curl --output scylla-gdb.py https://raw.githubusercontent.com/scylladb/${PRODUCT}/${MAIN_BRANCH}/scylla-gdb.py
```

Comment:
Let's use the one from the relocatable package (and maintain it).

Reply:
By maintaining it you mean backport fixes to `scylla-gdb.py` from now on? How about features?
Even if we started doing that today, there would be a 1-3 year window where some versions would have broken/outdated `scylla-gdb.py` versions.

Reply:
Hmm. I usually use the one from the relocatable package, but I likely open fewer cores.

The reason I prefer it is that it makes things self-contained and insulates us against breaking changes in scylla-gdb.py. But I understand the motivation for latest.

How about --fresh-scylla-gdb to pull the latest?

Reply:
Good idea. I'm trying to add flags so that the script can cater to any use-case.